# HG changeset patch
# User Andreas Weber <octave@josoansi.de>
# Date 1546552495 -3600
#      Thu Jan 03 22:54:55 2019 +0100
# Branch stable
# Node ID d644dec272ea2e2e9e9b49fb4dd0912f3b67c54e
# Parent  c67501d555418957ad25e5a8a0daa64e552bdbb6
ls-oct-text.cc: Fix static analyzer detected V519 issue (bug #55347).

diff --git a/libinterp/corefcn/ls-oct-text.cc b/libinterp/corefcn/ls-oct-text.cc
--- a/libinterp/corefcn/ls-oct-text.cc
+++ b/libinterp/corefcn/ls-oct-text.cc
@@ -303,35 +303,33 @@ read_text_data (std::istream& is, const 
 
 // FIXME: should probably write the help string here too.
 
 bool
 save_text_data (std::ostream& os, const octave_value& val_arg,
                 const std::string& name, bool mark_global,
                 int precision)
 {
-  bool success = true;
-
   if (! name.empty ())
     os << "# name: " << name << "\n";
 
   octave_value val = val_arg;
 
   if (mark_global)
     os << "# type: global " << val.type_name () << "\n";
   else
     os << "# type: " << val.type_name () << "\n";
 
   if (! precision)
     precision = Vsave_precision;
 
   long old_precision = os.precision ();
   os.precision (precision);
 
-  success = val.save_ascii (os);
+  bool success = val.save_ascii (os);
 
   // Insert an extra pair of newline characters after the data so that
   // multiple data elements may be handled separately by gnuplot (see
   // the description of the index qualifier for the plot command in the
   // gnuplot documentation).
   os << "\n\n";
 
   os.precision (old_precision);
